src/main/java/org/sonarlint/intellij/trigger/SonarDocumentListener.java

```diff
@@ -180,4 +180,24 @@ public void disposeComponent() {
     watcher.stopWatcher();
     eventMap.clear();
   }
+
+  /**
+   * Check if a file is currently open in the editor or not.
+   */
+  interface OpenFileChecker extends Function<VirtualFile, Boolean> {
```

Comment:
You could add isFileOpen(VirtualFile) to SonarLintAppUtils. It would allow to have all those utility functions together (that can be mocked) and avoid adding parameters and classes